npm/podController_test.go

```diff
@@ -0,0 +1,372 @@
+		if len(f.npMgr.PodMap) != 0 {
+			t.Errorf("addPod failed @ PodMap length check -current len %d %v", len(f.npMgr.PodMap), f.npMgr.PodMap)
+		}
+	} else {
+		if len(f.npMgr.PodMap) != 1 {
+			t.Errorf("addPod failed @ PodMap length check - current len %d %v", len(f.npMgr.PodMap), f.npMgr.PodMap)
+		}
+	}
+
+	if podObj.Spec.HostNetwork {
+		if len(f.npMgr.NsMap) != 1 {
+			t.Errorf("addPod failed @ nsMap length check - current len %d %v", len(f.npMgr.NsMap), f.npMgr.NsMap)
+		}
+	} else {
+		if len(f.npMgr.NsMap) != 2 {
+			//  all-namespaces and ns-test-namespace
+			t.Errorf("addPod failed @ nsMap length check - current len %d %v", len(f.npMgr.NsMap), f.npMgr.NsMap)
+		}
+	}
+}
+
+// // TODO: who call this function for message - type?
+func deletePod(t *testing.T, f *podFixture, podObj *corev1.Pod) {
+	addPod(t, f, podObj)
+	t.Logf("Complete add pod event")
+
+	// simulate pod delete event and delete pod object from sharedInformer cache
+	f.kubeInformer.Core().V1().Pods().Informer().GetIndexer().Delete(podObj)
+
+	// call deletePod event
+	f.podController.deletePod(podObj)
+	if !podObj.Spec.HostNetwork {
+		err := f.podController.syncPod(getKey(podObj, t))
+		if err != nil {
+			f.t.Errorf("error syncing pod: %v", err)
+		}
+	}
+
+	// TODO: add more checking list and clean up with tables with expected values.
+	if len(f.npMgr.PodMap) != 0 {
+		t.Errorf("deletePod failed @ PodMap length check -current len %d %v", len(f.npMgr.PodMap), f.npMgr.PodMap)
+	}
+
+	if podObj.Spec.HostNetwork {
+		if len(f.npMgr.NsMap) != 1 {
+			t.Errorf("deletePod failed for HostNetwork @ nsMap length check - current len %d %v", len(f.npMgr.NsMap), f.npMgr.NsMap)
+		}
+	} else {
+		if len(f.npMgr.NsMap) != 2 {
+			//  all-namespaces and ns-test-namespace
+			t.Errorf("deletePod failed @ nsMap length check - current len %d %v", len(f.npMgr.NsMap), f.npMgr.NsMap)
+		}
+	}
+}
+
+// Need to make more cases - interestings..
+func updatePod(t *testing.T, f *podFixture, oldPodObj *corev1.Pod, newPodObj *corev1.Pod) {
+	addPod(t, f, oldPodObj)
+	t.Logf("Complete add pod event")
+
+	// simulate pod update event and update the pod to shared informer's cache
+	f.kubeInformer.Core().V1().Pods().Informer().GetIndexer().Update(newPodObj)
+	f.podController.updatePod(oldPodObj, newPodObj)
+
+	// call deletePod event
+	if !newPodObj.Spec.HostNetwork {
+		err := f.podController.syncPod(getKey(newPodObj, t))
+		if err != nil {
+			f.t.Errorf("error syncing pod: %v", err)
+		}
+	}
+
+	// simulate pod delete event and delete pod object from sharedInformer cache
+	f.kubeInformer.Core().V1().Pods().Informer().GetIndexer().Delete(oldPodObj)
+
+	// call deletePod event
+	f.podController.deletePod(oldPodObj)
+	if !oldPodObj.Spec.HostNetwork {
+		err := f.podController.syncPod(getKey(oldPodObj, t))
+		if err != nil {
+			f.t.Errorf("error syncing pod: %v", err)
+		}
+	}
+
+	// TODO: add more checking list and clean up with tables with expected values.
+	if newPodObj.Spec.HostNetwork {
```

Comment:
Since this is being check multiple times, we can have check function like below

func checkCachemapStatus(maplen = len(Pod/NS map), newPodObj.Spec.HostNetwork, a=1, b=2) {
switch newPodObj.Spec.HostNetwork {
case true:
  if maplen != a
 return
case false:
 if maplen != b
retrun
}
}

Reply:
Applied with the ideas.